### PR TITLE
fix(gomodifytags): make gomodifytags more flexible

### PR DESCRIPTION
**Pull request by vanhtuan0409**
# Context

`gomodifytags` code actions currently hard coded binary path and does don't support customization from `generator_opts`

# Solution

Make a simple `factory` wrapper around current `generator` object to accept resolved generator opts

This PR does not change behavior of the current code actions. Only add additional layer for customization

### CODE REVIEW

File: lua/null-ls/builtins/code_actions/gomodifytags.lua [hunk 1]
```diff
@@ -4,10 +4,10 @@ local methods = require("null-ls.methods")
 
 local CODE_ACTION = methods.internal.CODE_ACTION
 
-local BIN = "gomodifytags -w -quiet "
-
-local exec = function(bufname, struct_name, field_name, op, tag)
+local exec = function(generator_opts, bufname, struct_name, field_name, op, tag)
     local opts = {}
+    table.insert(opts, "-w")
```
Comment (jose-elias-alvarez) [inline lua/null-ls/builtins/code_actions/gomodifytags.lua:9]: We can use `vim.list_extend()` here to create a single table that contains the command and the necessary arguments, then `concat` it into a single string.

**Review by jose-elias-alvarez — COMMENTED** [review]
This is an interesting use of `factory` that I hadn't thought of. I like that users don't have to learn an additional interface to modify the command, and this may be a good pattern for other command-based sources that don't use null-ls' spawning mechanism. Thanks!

File: lua/null-ls/builtins/code_actions/gomodifytags.lua (hunk 1 above)
Comment (vanhtuan0409) [reply lua/null-ls/builtins/code_actions/gomodifytags.lua:9]: Yeah right, i just keep the original author code here. Lets me see if I can make it cleaner

File: lua/null-ls/builtins/code_actions/gomodifytags.lua [hunk 2]
```diff
@@ -4,10 +4,10 @@ local methods = require("null-ls.methods")
 
 local CODE_ACTION = methods.internal.CODE_ACTION
 
-local BIN = "gomodifytags -w -quiet "
-
-local exec = function(bufname, struct_name, field_name, op, tag)
+local exec = function(generator_opts, bufname, struct_name, field_name, op, tag)
```
[the remark this answers is not in the document]
Comment (vanhtuan0409) [reply lua/null-ls/builtins/code_actions/gomodifytags.lua:7]: This is actually an intesting idea. Lets me do it

File: lua/null-ls/builtins/code_actions/gomodifytags.lua [hunk 3]
```diff
@@ -104,57 +104,62 @@ return h.make_builtin({
...
-
-            -- struct field
-            if (tsnode:type()) == "field_identifier" then
-                local field_name = vim.treesitter.query.get_node_text(tsnode, 0)
-                local tspnode = tsnode:parent():parent():parent()
-                if tspnode ~= nil and (tspnode:type()) == "struct_type" then
-                    tspnode = tspnode:parent():child(0)
-                    struct_name = vim.treesitter.query.get_node_text(tspnode, 0)
+    generator_opts = {
+        command = "gomodifytags",
```
[the remark this answers is not in the document]
Comment (vanhtuan0409) [reply lua/null-ls/builtins/code_actions/gomodifytags.lua:108]: The command actually varied based on the action selected. I can use the args as an extra flags added to each action. How do you think

**Comment by vanhtuan0409** [comment]
all resolved. Could you take a look again @jose-elias-alvarez

**Comment by jose-elias-alvarez** [comment]
Rebased to fix the CI failure, and other than that this looks good. Thanks!

**Pull request merged**